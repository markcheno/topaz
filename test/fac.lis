#    0: adjs		1
#    5: jmp		129
#   10: link		1
#   15: getlocal	-2
#   20: pushint		0
#   25: eql
#   26: jz		51
#   31: pushint		1
#   36: setlocal	-3
#   41: rts		3
#   46: jmp		51
#   51: pushint		1
#   56: setlocal	0
#   61: getlocal	-2
#   66: pushint		0
#   71: gtr
#   72: jz		114
#   77: getlocal	0
#   82: getlocal	-2
#   87: mul
#   88: setlocal	0
#   93: getlocal	-2
#   98: pushint		1
#  103: sub
#  104: setlocal	-2
#  109: jmp		61
#  114: getlocal	0
#  119: setlocal	-3
#  124: rts		3
#  129: pushint		7
#  134: setglobal	0
#  139: pushint		0
#  144: pushstring	'factorial of '
#  163: getglobal	0
#  168: pushstring	' = '
#  177: pushint		0
#  182: getglobal	0
#  187: pushint		1
#  192: jsr		10
#  197: pushint		4
#  202: call		println
#  207: pop
#  208: halt
