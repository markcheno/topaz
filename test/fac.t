# Calculate Factorial of n
def fact(n)

  if n == 0 then return 1 end
  
  f=1
  while n > 0 do
    f = f*n
    n = n-1
  end
  return f
  
end

x = 7

println("factorial of ",x," = ",fact(x))
