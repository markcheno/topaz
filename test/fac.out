factorial of 7 = 5040
